pkg/code-generator/codegen/generator.go

```diff
@@ -36,7 +36,7 @@ func GenerateFiles(project *model.Project, skipOutOfPackageFiles bool) (code_gen
 		files = append(files, fs...)
 	}
 	for _, grp := range project.ResourceGroups {
-		if skipOutOfPackageFiles && !strings.HasSuffix(grp.Name, "."+project.GroupName) {
+		if skipOutOfPackageFiles && !(strings.HasSuffix(grp.Name, "."+project.GroupName) || grp.Name == project.GroupName) {
```

Comment:
I expected this to become `!strings.HasSuffix(grp.Name, project.GroupName)` 

Works fine according to https://golang.org/pkg/strings/#HasSuffix

Reply:
this is problematic if project group names themselves have overlaps. e.g. `supergloo.solo.io` and `gloo.solo.io`
that's why we need the prefix case for `.gloo.solo.io` as well as `==` case for `gloo.solo.io`